airflow/models/queue_task_run.py

```diff
@@ -0,0 +1,123 @@
+        and trigger rule
+    :type ignore_dependencies: bool
+    :param force: Ignore the task instance's previous failure/success
+    :type force: bool
+    :param local: Whether to run the task locally
+    :type local: bool
+    :param pickle_id: If the DAG was serialized to the DB, the ID
+        associated with the pickled DAG
+    :type pickle_id: unicode
+    :param subdir: path to the file containing the DAG definition
+    :param raw: raw mode (needs more details)
+    :param job_id: job ID (needs more details)
+    :param pool: the Airflow pool that the task should run in
+    :type pool: unicode
+    :param cfg_path: the Path to the configuration file
+    :type cfg_path: str
+    :return: shell command that can be used to run the task instance
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        dag_id,
+        task_id,
+        execution_date,
+        mark_success=None,
+        pickle_id=None,
+        job_id=None,
+        ignore_all_dependencies=None,
+        ignore_dependencies=None,
+        ignore_depends_on_past=None,
+        force=None,
+        local=None,
+        pool=None,
+        raw=None,
+        subdir=None,
+        cfg_path=None
+    ):
+        self.dag_id = dag_id
+        self.task_id = task_id
+        self.execution_date = execution_date
+        self.mark_success = mark_success
+        self.pickle_id = pickle_id
+        self.job_id = job_id
+        self.ignore_all_dependencies = ignore_all_dependencies
+        self.ignore_dependencies = ignore_dependencies
+        self.ignore_depends_on_past = ignore_depends_on_past
+        self.force = force
+        self.local = local
+        self.pool = pool
+        self.raw = raw
+        self.subdir = subdir
+        self.cfg_path = cfg_path
+
+    def as_command(self):
+        """Generate CLI command"""
+        iso = self.execution_date.isoformat()
+        cmd = ["airflow", "tasks", "run", str(self.dag_id), str(self.task_id), str(iso)]
+        if self.mark_success:
+            cmd.extend(["--mark_success"])
+        if self.pickle_id:
+            cmd.extend(["--pickle", str(self.pickle_id)])
+        if self.job_id:
+            cmd.extend(["--job_id", str(self.job_id)])
+        if self.ignore_all_dependencies:
+            cmd.extend(["--ignore_all_dependencies"])
+        if self.ignore_dependencies:
+            cmd.extend(["--ignore_dependencies"])
+        if self.ignore_depends_on_past:
+            cmd.extend(["--ignore_depends_on_past"])
+        if self.force:
+            cmd.extend(["--force"])
+        if self.local:
+            cmd.extend(["--local"])
+        if self.pool:
+            cmd.extend(["--pool", self.pool])
+        if self.raw:
+            cmd.extend(["--raw"])
+        if self.subdir:
+            cmd.extend(["--subdir", self.subdir])
+        if self.cfg_path:
+            cmd.extend(["--cfg_path", self.cfg_path])
+        return cmd
+
```

Comment:
Awesome! But we still pass the "command". I would like to abandon this "command build" approach and use a dictionary / object that will be accepted by `task_command.task_run`.  

Probably this will mean that we have to extract some common logic from `task_command.task_run`. WDYT?

Reply:
I agree, but I would prefer to do it step by step.

Reply:
Definitely!